tests/Type/QueryPlanTest.php

```diff
@@ -0,0 +1,588 @@
+                                'args' => [],
+                                'fields' => [
+                                    'id' => [
+                                        'type' => Type::string(),
+                                        'args' => [],
+                                        'fields' => [],
+                                    ],
+                                    'title' => [
+                                        'type' => Type::string(),
+                                        'args' => [],
+                                        'fields' => [],
+                                    ],
+                                    'body' => [
+                                        'type' => Type::string(),
+                                        'args' => [],
+                                        'fields' => [],
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $expectedReferencedTypes = [
+            'Image',
+            'Author',
+            'Article',
+            'Reply',
+        ];
+
+        $expectedReferencedFields = [
+            'url',
+            'width',
+            'height',
+            'name',
+            'pic',
+            'id',
+            'recentArticle',
+            'title',
+            'body',
+            'author',
+            'image',
+            'replies',
+        ];
+
+        $hasCalled = false;
+        /** @var QueryPlan $queryPlan */
+        $queryPlan = null;
+
+        $blogQuery = new ObjectType([
+            'name'   => 'Query',
+            'fields' => [
+                'article' => [
+                    'type'    => $article,
+                    'resolve' => static function (
+                        $value,
+                        $args,
+                        $context,
+                        ResolveInfo $info
+                    ) use (
+                        &$hasCalled,
+                        &$queryPlan
+                    ) {
+                        $hasCalled = true;
+                        $queryPlan = $info->lookahead();
+
+                        return null;
+                    },
+                ],
+            ],
+        ]);
+
+        $schema = new Schema(['query' => $blogQuery]);
+        $result = GraphQL::executeQuery($schema, $doc)->toArray();
+
+        self::assertTrue($hasCalled);
+        self::assertEquals(['data' => ['article' => null]], $result);
+        self::assertEquals($expectedQueryPlan, $queryPlan->queryPlan());
+        self::assertEquals($expectedReferencedTypes, $queryPlan->referencedTypes());
+        self::assertEquals($expectedReferencedFields, $queryPlan->referencedFields());
+        self::assertEquals(['url', 'width', 'height'], $queryPlan->subFields('Image'));
+
+        static::assertTrue($queryPlan->hasField('url'));
```

Comment:
What's up with the `self::` vs. `static::` calls here?

Maybe it's through out the codebase and I just didn't notice it before 🤔